azurelinuxagent/common/protocol/wire.py

```diff
@@ -913,49 +913,51 @@ def _call_hostplugin_with_container_check(self, host_func):
+            HostPluginProtocol.set_default_channel(True)
 
-        # NOTE: direct_func and host_func are passed as lambdas. Be careful about capturing goal state data in them as
-        # they will not be refreshed even if a goal state refresh is called before retrying the host_func.
+        return ret
 
-        if not HostPluginProtocol.is_default_channel():
-            ret = None
-            try:
-                ret = direct_func()
+    def __send_request_using_direct_channel(self, direct_func):
+        ret = None
+        try:
+            ret = direct_func()
 
-                # Different direct channel functions report failure in different ways: by returning None, False,
-                # or raising ResourceGone or InvalidContainer exceptions.
-                if not ret:
-                    logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
-                                                            "switching to host plugin.")
-            except (ResourceGoneError, InvalidContainerError) as e: # pylint: disable=C0103
+            # Different direct channel functions report failure in different ways: by returning None, False,
+            # or raising ResourceGone or InvalidContainer exceptions.
+            if not ret:
                 logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
-                                                        "switching to host plugin. Error: {0}".format(ustr(e)))
+                                                        "trying host plugin next.")
+        except (ResourceGoneError, InvalidContainerError) as e:  # pylint: disable=C0103
+            logger.periodic_info(logger.EVERY_HOUR, "[PERIODIC] Request failed using the direct channel, "
+                                                    "trying host plugin next. Error: {0}".format(ustr(e)))
 
-            if ret:
-                return ret
-        else:
-            logger.periodic_info(logger.EVERY_HALF_DAY, "[PERIODIC] Using host plugin as default channel.")
+        if ret and HostPluginProtocol.is_default_channel():
+            # Set direct channel as default if it succeeded.
+            logger.info("Switching default channel to WireServer.")
+            HostPluginProtocol.set_default_channel(False)
 
-        ret = self._call_hostplugin_with_container_check(host_func)
+        return ret
 
-        if not HostPluginProtocol.is_default_channel():
-            logger.info("Setting host plugin as default channel from now on. "
-                        "Restart the agent to reset the default channel.")
-            HostPluginProtocol.set_default_channel(True)
+    def send_request_using_appropriate_channel(self, direct_func, host_func):
+        primary_ret = None
+        try:
+            primary_ret = self.__send_request_using_host_channel(host_func) if HostPluginProtocol.is_default_channel() \
```

Comment:
Would a verbose message indicating the channel used be too verbose and fill up the logs? Or is there already a log entry that does something similar when requests are successful?

Reply:
I was afraid it would be too verbose. HostPlugin will have logs if guest agent used it so this comment can be resolved.